beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/chains/SyncSourceFactory.java

```diff
@@ -44,15 +44,20 @@ public SyncSourceFactory(
   public SyncSource getOrCreateSyncSource(final Eth2Peer peer, final Spec spec) {
     // Limit request rate to just a little under what we'd accept
     final int maxBlocksPerMinute = this.maxBlocksPerMinute - batchSize - 1;
+    final Optional<Integer> maybeMaxBlobsPerBlock = spec.getMaxBlobsPerBlockForHighestMilestone();
```

Comment:
so if electra goes to 9 and hasn't been activated we'd still use electra's max blobs?

Reply:
Yeah, this one I have to talk about Enrico what is the best strategy. We pre-initialize the RateTracker on startup, so it's not dynamic per milestone.

Reply:
ye this is a little bit of a pain. At the end in this cases I think it is an acceptable side-effect but we could be rate-limited more during the transition period. We could go this simplified way and rise an issue describing the issue, so we can later think about more correct ways

Reply:
I think there is not that much benefit, keep in mind we `adjust` the number after getting `x` number of blobs and we are the default blobs per minute will still be the same for future upgrades, so all in all not much reason to complicate the code for little benefit in my opinion

Reply:
In the DefaultEth2Peer, the RateTracker just use the defaults. Only in `SyncSourceFactory`, we modify the number slightly to make sure we don't rate limit ourselves.